pkg/deploy/strategy/recreate/recreate.go

```diff
@@ -13,52 +14,50 @@ import (
 	deployutil "github.com/openshift/origin/pkg/deploy/util"
 )
 
-// DeploymentStrategy is a simple strategy appropriate as a default. Its behavior is to increase the
+// RecreateDeploymentStrategy is a simple strategy appropriate as a default. Its behavior is to increase the
 // replica count of the new deployment to 1, and to decrease the replica count of previous deployments
 // to zero.
 //
 // A failure to disable any existing deployments will be considered a deployment failure.
-type DeploymentStrategy struct {
-	// ReplicationController is used to interact with ReplicatonControllers.
-	ReplicationController ReplicationControllerInterface
-	// Codec is used to decode DeploymentConfigs contained in deployments.
-	Codec runtime.Codec
+type RecreateDeploymentStrategy struct {
+	// client is used to interact with ReplicatonControllers.
+	client replicationControllerClient
+	// codec is used to decode DeploymentConfigs contained in deployments.
+	codec runtime.Codec
+
+	retryTimeout time.Duration
+	retryPeriod  time.Duration
+}
+
+func NewRecreateDeploymentStrategy(client kclient.Interface, codec runtime.Codec) *RecreateDeploymentStrategy {
+	return &RecreateDeploymentStrategy{
+		client:       &realReplicationController{client},
+		codec:        codec,
+		retryTimeout: 10 * time.Second,
+		retryPeriod:  1 * time.Second,
```

Comment:
I made the retry logic continue immediately when `IsConflict`. Timeout is still effective, though. Seems like if there's enough thrashing on an RC that you can't get an update in with tight-loop attempts over 10 seconds, something somewhere's probably very wrong and we should fail anyway. What do you think?